### PR TITLE
[Cosmos] Include new sequence field next to signature

### LINKED ISSUE #2089: [Thorchain] Cosmos v0.45 SDK upgrade
Thorchain disables standard Cosmos `cosmos-sdk/MsgSend` and uses its own `thorchain/MsgSend` https://gitlab.com/thorchain/thornode/-/blob/develop/proto/thorchain/v1/x/thorchain/types/msg_send.proto

```protobuf
syntax = "proto3";
package types;

option go_package = "gitlab.com/thorchain/thornode/x/thorchain/types";

import "cosmos/base/v1beta1/coin.proto";
import "gogoproto/gogo.proto";

message MsgSend {
  bytes from_address = 1  [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
  bytes to_address = 2  [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
  repeated cosmos.base.v1beta1.Coin amount = 3 [(gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins", (gogoproto.nullable) = false];
}
```

The difference is: `from_address` and `to_address` type is `bytes` not `string`

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

- Json-format tx is deprecated, Protobuf-based (Stargate) should be used.  This change affects only Json format.
- Triggered by ThorChain, breaking change, Cosmos SDK 0.45
- Sequence value is echoed in the output, next to the signature field.
- Backward compatible change, only an additional field
- Change affects all Cosmos-based chains, but in Json-serialization only.

Fixes #2089 .

More info:
https://lively-router-cb2.notion.site/Cosmos-v0-45-Migration-Guide-for-THORChain-Interfaces-1d140176017b433da79f77200904fbb6

## Testing instructions

Could not be validated in a live chain, as Thorchain, Cosmos, Cryptoorg no longer supports Json-serialization.
Unit tests updated.

## Types of changes

* Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

**Comment by hewigovens** [comment]
Let's support new protobuf format first? App will migrate to protobuf mode in next build

**Comment by catenocrypt** [comment]
Note: More important is to support new stargate-serialized messages for Thorchain transfer, being done first in #2092 .

File: tests/THORChain/SignerTests.cpp [hunk 1]
```diff
@@ -185,13 +187,14 @@ TEST(THORChainSigner, SignTx_Json_Deprecated) {
...
                             "value": "A+2Zfjls9CkvX85aQrukFZnM1dluMTFUp8nqcEneMXx3"
                         },
-                        "signature": "12AaNC0v51Rhz8rBf7V7rpI6oksREWrjzba3RK1v1NNlqZq62sG0aXWvStp9zZXe07Pp2FviFBAx+uqWsO30NQ=="
+                        "signature": "4rE015+9PfMPmylMiPuE0OSCEE8vfgoMbH2AFTfyqU4t09kRNeH1fAV8c4yQ3CxB+0gUBySZrM0iygA3626v8w==",
+                        "sequence": 123
                     }
                 ]
             }
-        }
+        }
```
Comment (hewigovens) [inline tests/THORChain/SignerTests.cpp:195]:
```suggestion
        }
```

**Review by hewigovens — COMMENTED** [review]
The new field looks no harm to other cosmos chains, better to verify it with one tx

**Comment by hewigovens** [comment]
@catenocrypt Let's remove json support? it seems not worth to maintain it, what cosmos chain is still using old sdk?

**Comment by hewigovens** [comment]
Let me check Band and Bluzelle, if those two already support stargate, we can probably remove json support

**Comment by hewigovens** [comment]
Band supports protobuf format, so we can just wait for Bluzelle then we can remove json support later

**Pull request closed without merging**